src/Utils.cpp

```diff
@@ -60,70 +65,69 @@ void CUtils::GenerateCert(FILE *pOut, const CString& sHost) {
+#endif
+	{ // <-- deviating from style so editors correctly match braces
+		if ((pKey = EVP_PKEY_new()) && EVP_PKEY_assign_RSA(pKey, pRSA)) {
+			PEM_write_RSAPrivateKey(pOut, pRSA, NULL, NULL, 0, NULL, NULL);
 
-		PEM_write_RSAPrivateKey(pOut, pRSA, NULL, NULL, 0, NULL, NULL);
+			if ((pCert = X509_new())) {
+				X509_set_version(pCert, 2);
+				ASN1_INTEGER_set(X509_get_serialNumber(pCert), serial);
+				X509_gmtime_adj(X509_get_notBefore(pCert), 0);
+				X509_gmtime_adj(X509_get_notAfter(pCert), (long)60*60*24*days*years);
+				X509_set_pubkey(pCert, pKey);
 
-		if (!(pCert = X509_new())) {
-			EVP_PKEY_free(pKey);
-			return;
-		}
+				pName = X509_get_subject_name(pCert);
 
-		X509_set_version(pCert, 2);
-		ASN1_INTEGER_set(X509_get_serialNumber(pCert), serial);
-		X509_gmtime_adj(X509_get_notBefore(pCert), 0);
-		X509_gmtime_adj(X509_get_notAfter(pCert), (long)60*60*24*days*years);
-		X509_set_pubkey(pCert, pKey);
+				const char *pLogName = getenv("LOGNAME");
+				const char *pHostName = NULL;
 
-		pName = X509_get_subject_name(pCert);
-
-		const char *pLogName = getenv("LOGNAME");
-		const char *pHostName = NULL;
+				if (!sHost.empty()) {
+					pHostName = sHost.c_str();
+				}
 
-		if (!sHost.empty()) {
-			pHostName = sHost.c_str();
-		}
+				if (!pHostName) {
+					pHostName = getenv("HOSTNAME");
+				}
 
-		if (!pHostName) {
-			pHostName = getenv("HOSTNAME");
-		}
+				if (!pLogName) {
+					pLogName = "Unknown";
+				}
 
-		if (!pLogName) {
-			pLogName = "Unknown";
-		}
+				if (!pHostName) {
+					pHostName = "host.unknown";
+				}
 
-		if (!pHostName) {
-			pHostName = "host.unknown";
-		}
+				CString sEmailAddr = pLogName;
+				sEmailAddr += "@";
+				sEmailAddr += pHostName;
 
-		CString sEmailAddr = pLogName;
-		sEmailAddr += "@";
-		sEmailAddr += pHostName;
+				X509_NAME_add_entry_by_txt(pName, "OU", MBSTRING_ASC, (unsigned char *)pLogName, -1, -1, 0);
+				X509_NAME_add_entry_by_txt(pName, "CN", MBSTRING_ASC, (unsigned char *)pHostName, -1, -1, 0);
+				X509_NAME_add_entry_by_txt(pName, "emailAddress", MBSTRING_ASC, (unsigned char *)sEmailAddr.c_str(), -1, -1, 0);
 
-		X509_NAME_add_entry_by_txt(pName, "OU", MBSTRING_ASC, (unsigned char *)pLogName, -1, -1, 0);
-		X509_NAME_add_entry_by_txt(pName, "CN", MBSTRING_ASC, (unsigned char *)pHostName, -1, -1, 0);
-		X509_NAME_add_entry_by_txt(pName, "emailAddress", MBSTRING_ASC, (unsigned char *)sEmailAddr.c_str(), -1, -1, 0);
+				X509_set_subject_name(pCert, pName);
+				X509_set_issuer_name(pCert, pName);
 
-		X509_set_subject_name(pCert, pName);
-		X509_set_issuer_name(pCert, pName);
+				if (X509_sign(pCert, pKey, EVP_sha256())) {
+					PEM_write_X509(pOut, pCert);
 
-		if (!X509_sign(pCert, pKey, EVP_sha256())) {
+					fprintf(pOut, "%s", szDefaultDH2048);
+				}
+			}
 			X509_free(pCert);
-			EVP_PKEY_free(pKey);
-			return;
 		}
-
-		PEM_write_X509(pOut, pCert);
-		X509_free(pCert);
 		EVP_PKEY_free(pKey);
-
-		fprintf(pOut, "%s", szDefaultDH2048);
 	}
+	RSA_free(pRSA);
```

Comment:
I'm not familiar with that syntax, but it looks fairly self explanatory. Assuming I can simply replace `RSA *pRSA = NULL;` with that and get rid of the later `RSA_free(pRSA);` line, should I do the same for other pointers that need freeing while I'm here?